pennylane/workflow/construct_batch.py

```diff
@@ -81,6 +83,110 @@ def _get_full_transform_program(
     return program + qnode.device.preprocess_transforms(config)
 
 
+def _level_sanity_check(
+    level: Optional[Literal["top", "user", "device", "gradient"] | int | slice],
+) -> None:
+    """Check that the level specification is valid.
+
+    Args:
+        level: The level specification from user input
+
+    Raises:
+        ValueError: If the level is not recognized
+    """
+    if level is None or isinstance(level, int):
+        return
+
+    if isinstance(level, str):
+        if level not in ("top", "user", "device", "gradient"):
+            raise ValueError(
+                f"level {level} not recognized. Acceptable strings are 'device', 'top', 'user', and 'gradient'."
+            )
+        return
+
+    if isinstance(level, slice):
+        return
+
+    raise ValueError(
+        f"level {level} not recognized. Acceptable types are None, int, str, and slice."
+    )
+
+
+def _interpret_level_initial(
```

Comment:
Another suggestion for readability (will need to propagate the change down if you agree),
```suggestion
def _get_user_transform_slice(
```

Reply:
https://github.com/PennyLaneAI/pennylane/pull/7461/commits/c6a4f3dc40677ade7314a10c80d551659b24567f